security-utils/src/main/java/com/yahoo/security/tls/PeerAuthorizer.java

```diff
@@ -35,9 +35,7 @@ public PeerAuthorizer(AuthorizedPeers authorizedPeers) {
     public ConnectionAuthContext authorizePeer(X509Certificate cert) { return authorizePeer(List.of(cert)); }
 
     public ConnectionAuthContext authorizePeer(List<X509Certificate> certChain) {
-        if (authorizedPeers.isEmpty()) {
-            return new ConnectionAuthContext(certChain, CapabilitySet.all(), Set.of());
-        }
+        if (authorizedPeers.isEmpty()) return ConnectionAuthContext.defaultAllCapabilities();
```

Comment:
Consider if we also want to expose the peer certificate information in `ConnectionAuthContext` even if it has no authz rules and is implicitly granted all capabilities